common/NativeObject.java

```diff
@@ -19,31 +19,16 @@
  * under the License.
  */
 
-package com.vaticle.typedb.client.api.query;
+package com.vaticle.typedb.client.common;
 
-import java.util.function.Function;
-
-public interface QueryFuture<T> {
-
-    T get();
-
-    default <U> QueryFuture<U> map(Function<T, U> function) {
-        return new Mapped<>(this, function);
+public abstract class NativeObject<T> {
+    static {
+        System.loadLibrary("typedb_client_jni");
     }
```

Comment:
Common superclass for any class that has a native representation. This ensures that the shared library is loaded before we use any of them.